app/views/shared/_navbar.html.erb

```diff
@@ -0,0 +1,24 @@
+<nav class="navbar is-fixed-top" role="navigation" aria-label="main navigation">
+  <div id="navbarBasicExample" class="navbar-menu">
+    <div class="navbar-start">
+      <% if user_signed_in? %>
+        <%= link_to "Food", foods_path, class: "navbar-item has-text-link" %>
+        <%= link_to "Recipes", recipes_path, class: "navbar-item has-text-link" %>
+        <%= link_to "Public recipes", recipes_path, class: "navbar-item has-text-link" %>
+        <%= link_to "Shopping list", recipes_path, class: "navbar-item has-text-link" %>
+      <% else %>
+        <%= link_to "Sign up", new_user_registration_path %> |
+        <%= link_to "Login", new_user_session_path %>
+      <% end %>
+    </div>
+
+    <div class="navbar-end">
+      <div class="navbar-item">
+        <span class="mr-3 has-text-success-dark">Logged in as: <strong><%= current_user.email %></strong></span>
+        <div class="buttons">
+          <%= button_to "Log out", destroy_user_session_path, method: :delete, class: "button is-small is-danger is-outlined" %>
+        </div>
```

Comment:
- [ ] We don't need to log out before a user login in or signs in.

![rails12](https://user-images.githubusercontent.com/55635977/142575698-1f6b6142-b51e-446b-95d3-809ba6666cf3.png)

Reply:
Fixed